src/pages/question/panels/EdgePanel.jsx

```diff
@@ -15,31 +22,76 @@ const listItem = ({ item }) => (
 );
 
 export default function EdgePanel(props) {
+  const predicateStatus = usePageStatus(false);
   const { panelStore } = props;
   const { edge } = panelStore;
 
-  /**
-   * Cancel any calls in progress
-   */
-  // useEffect(() => () => {
-  //   edge.cancelConnectionsCall();
-  //   edge.cancelPredicatesCall();
-  // }, []);
+  async function fetchPredicates() {
+    if (!edge.sourceId || !edge.targetId) {
+      return;
+    }
+    const sourceNode =
+      { ...panelStore.query_graph.nodes[edge.sourceId] };
+    const targetNode =
+      { ...panelStore.query_graph.nodes[edge.targetId] };
+      // Including name or empty curie list in the node breaks the API call
+      // So we make a copy and remove it
+    delete sourceNode.name;
+    delete targetNode.name;
+    if (sourceNode.curie.length === 0) delete sourceNode.curie;
+    if (targetNode.curie.length === 0) delete targetNode.curie;
```

Comment:
Sorry. We also need to handle when a node doesn't have a curie. A node could be as simple as { type: 'gene' }.